### ISSUE TITLE
[New Resource] QoS Policies

### ISSUE THREAD

**Issue by tysonjhayes** [PowerShell/xNetworking#77]
Add  [QoS Policy](https://technet.microsoft.com/en-us/library/hh967469%28v=wps.630%29.aspx) to the list of resources.

**Comment by rchaganti** [comment]
Got a couple of resources here.

1. MSFT_xNetworkAdapterQos - Enables/Disables QoS on a network adapter.
2. MSFT_xNetQosTrafficClass - Manage traffic classes creation and deletion, algorithm / priority management and so on.

but, the question I have is, should these be in xNetworking or go into a module of their own like NetworkQoSDsc?

**Comment by PlagueHO** [comment]
Hi @rchaganti - good question.

Can you forsee the need for any other Network QoS resources? If there is likely to be a whole load of them it might be better to keep them in a new Resource Module. But if these two are all there are likely to be if might be better to include them in xNetworking (one day to be NetworkingDsc I hope).

Ultimately it should be down to discoverability - e.g. where would you expect to find them if you were looking. Because the cmdlets that configure QoS are *-Net* cmdlets I'd first go looking for the resources in the Networking repo. But I guess if they are in the NetworkQoSDsc that is equally discoverable. But because xNetworking is fairly well known/used, people might go looking there first.

@tysonjhayes, @kwirkykat, @mbreakey3, @TravisEz13 ?

**Comment by rchaganti** [comment]
What I am looking at is a larger collection that spans across three different modules.

`Get-Command -Module NetQoS

Get-NetQosPolicy
New-NetQosPolicy
Remove-NetQosPolicy
Set-NetQosPolicy

Get-Command -Module NetAdapter -Noun *QoS*

Disable-NetAdapterQos 
Enable-NetAdapterQos
Get-NetAdapterQos
Set-NetAdapterQos 

Get-Command -Module dcBqos

Disable-NetQosFlowControl
Enable-NetQosFlowControl
Get-NetQosDcbxSetting 
Get-NetQosFlowControl 
Get-NetQosTrafficClass        
New-NetQosTrafficClass      
Remove-NetQosTrafficClass
Set-NetQosDcbxSetting   
Set-NetQosFlowControl   
Set-NetQosTrafficClass     
Switch-NetQosDcbxSetting 
Switch-NetQosFlowControl 
Switch-NetQosTrafficClass`

While these are physically separate, they are all used in a QoS Configuration for example in a S2D cluster.  I have the MSFT_NetAdapterQoS and MSFT_NetQosTrafficClass in a xNetworking fork but wanted to see community thoughts on the grouping. I am working on MSFT_NetQoSDcbxSetting, MSFT_NetQosPolicy, and MSFT_NetQosFlowControl. So, it will be total 5 resources.

**Comment by PlagueHO** [comment]
@rchaganti - in that case, maybe a separate QoS resource module might be the way to go. But would be good to get some thoughts from the MSFT team.

**Comment by kwirkykat** [comment]
@rchaganti @PlagueHO Yeah looking at that list of functions, NetQos or NetworkQos should be its own resource module.

**Comment by BarnumD** [comment]
Interested in using these..

**Comment by PlagueHO** [comment]
@rchaganti - did you ed up implementing or releasing any NetQos DSC Resources?

**Comment by rchaganti** [comment]
Yes. I implemented a bunch of these resources (NetQosDSC) and will publish my repo this week. I will need help in writing some tests though. I don't have the full coverage.

**Comment by PlagueHO** [comment]
Thanks @rchaganti - awesome! :smile: Could you link to them in this issue when they're published? I'd contribute some tests for you but I'm miles behind in the backlog for the other repos I maintain! :cry:

**Comment by rchaganti** [comment]
Alright. Here it is: https://github.com/rchaganti/netqosdsc
I added some basic examples and no tests at all. I am adding tests but cannot publish them until holidays are over.
